lib/spack/llnl/util/cpu/microarchitectures.json

```diff
@@ -1161,6 +1161,97 @@
         }
       }
     },
+    "thunderx2": {
+      "from": "aarch64",
+      "vendor": "0x43",
```

Comment:
Apologies for arriving late at this. Are we keeping vendor names non human-readable? This will be displayed as `0x43 - aarch64` when people ask for:
```console
$ spack arch --known-targets
```

Reply:
Well... `CPU implementer` in `/proc/cpuinfo` is only displayed with specific number as vendor information. So, I implemented `vendor` with numbers. Actually, `0x43` means `Cavium`, it is vendor name. (I will check once about `0x46`)

I check `$ spack arch --known-targets`, and confirm following.
```
0x43 - aarch64
    thunderx2

0x46 - aarch64
    a64fx
```
Is there way of changing vendor name displayed with this command?

Reply:
> Well... CPU implementer in /proc/cpuinfo is only displayed with specific number as vendor information. So, I implemented vendor with numbers.

No worries, was wondering if we want to map the code to the name. Probably it makes sense to do it when we detect the raw information in more or less the same way that we do to map a few instructions sets from `Darwin` to what we currently have in the JSON file:

https://github.com/spack/spack/blob/66cf530459237573597cda6280f44f464ce1326b/lib/spack/llnl/util/cpu/detect.py#L114-L123

Reply:
@t-karatsu Do you have a recent version of `lscpu` from `util-linux` installed on your system? I am wondering if they do this mapping...

Reply:
Oh, thanks! We will consider implementations of vendor name mapping.
>@t-karatsu Do you have a recent version of lscpu from util-linux installed on your system? I am wondering if they do this mapping...

Some of the thunderX2 systems we are able to use, show vendor names(Cavium) using the lscpu command.

Reply:
Thanks for confirming @t-karatsu! I'll search into their code and see if I can find where they keep the mapping. We might be able to reuse that.

Reply:
@becker33 @t-karatsu @tgamblin This file can be interesting for us: [lscpu-arm.c](https://github.com/karelzak/util-linux/blob/master/sys-utils/lscpu-arm.c). The part that is most relevant to this discussion is:
```c
static const struct hw_impl hw_implementer[] = {
    { 0x41, arm_part,     "ARM" },
    { 0x42, brcm_part,    "Broadcom" },
    { 0x43, cavium_part,  "Cavium" },
    { 0x44, dec_part,     "DEC" },
    { 0x48, hisi_part,  "HiSilicon" },
    { 0x4e, nvidia_part,  "Nvidia" },
    { 0x50, apm_part,     "APM" },
    { 0x51, qcom_part,    "Qualcomm" },
    { 0x53, samsung_part, "Samsung" },
    { 0x56, marvell_part, "Marvell" },
    { 0x66, faraday_part, "Faraday" },
    { 0x69, intel_part,   "Intel" },
    { -1,   unknown_part, "unknown" },
};
```